File: Task_1/src/cpp_pubsub/src/publisher_node.cpp
#include "rclcpp/rclcpp.hpp"
#include "std_msgs/msg/string.hpp"
#include <chrono>

using namespace std::chrono_literals;

class CppPublisher : public rclcpp::Node {
public:
  CppPublisher() : Node("cpp_publisher"), count_(0) {
    publisher_ = create_publisher<std_msgs::msg::String>("cpp_topic", 10);
    timer_ = create_wall_timer(500ms, [this]() { timer_callback(); });
  }
private:
  void timer_callback() {
    auto msg = std_msgs::msg::String();
    msg.data = "Hello from C++: " + std::to_string(count_++);
    RCLCPP_INFO(get_logger(), "Publishing: '%s'", msg.data.c_str());
    publisher_->publish(msg);
  }
  rclcpp::Publisher<std_msgs::msg::String>::SharedPtr publisher_;
  rclcpp::TimerBase::SharedPtr timer_;
  size_t count_;
};

int main(int argc, char* argv[]) {
  rclcpp::init(argc, argv);
  rclcpp::spin(std::make_shared<CppPublisher>());
  rclcpp::shutdown();
  return 0;
}

File: Tasks/Task_1/src/py_pubsub/py_pubsub/publisher_node.py
import rclpy
from rclpy.node import Node
from std_msgs.msg import String

class PyPublisher(Node):
    def __init__(self):
        super().__init__('py_publisher')
        self.publisher_ = self.create_publisher(String, 'py_topic', 10)
        self.count = 0
        self.timer = self.create_timer(0.5, self.timer_callback)

    def timer_callback(self):
        msg = String()
        msg.data = f'Hello from Python: {self.count}'
        self.publisher_.publish(msg)
        self.get_logger().info(f'Publishing: "{msg.data}"')
        self.count += 1

def main(args=None):
    rclpy.init(args=args)
    node = PyPublisher()
    rclpy.spin(node)
    node.destroy_node()
    rclpy.shutdown()
